datastore/schedule-export/src/main/java/com/example/datastore/DatastoreExportServlet.java

```diff
@@ -0,0 +1,147 @@
+
+      // Get an access token to authorize export request
+      ArrayList<String> scopes = new ArrayList<String>();
+      scopes.add("https://www.googleapis.com/auth/datastore");
+      final AppIdentityService appIdentity = AppIdentityServiceFactory.getAppIdentityService();
+      final AppIdentityService.GetAccessTokenResult accessToken =
+          AppIdentityServiceFactory.getAppIdentityService().getAccessToken(scopes);
+      connection.addRequestProperty("Authorization", "Bearer " + accessToken.getAccessToken());
+
+      // Build export request payload based on URL parameters
+      // Required: output_url_prefix
+      // Optional: entity filter
+      JSONObject exportRequest = new JSONObject();
+
+      // If output prefix ends with a slash, use as is
+      // Otherwise, add a timestamp to form unique output url
+      if (!outputUrlPrefix.endsWith("/")) {
+        String timeStamp = new SimpleDateFormat("yyyyMMddHHmmss").format(new Date());
+        outputUrlPrefix = outputUrlPrefix + "/" + timeStamp + "/";
+      }
+
+      // Add outputUrl to payload
+      exportRequest.put("output_url_prefix", outputUrlPrefix);
+
+      // Build optional entity filter to export subset of
+      // kinds or namespaces
+      JSONObject entityFilter = new JSONObject();
+
+      // Read kind parameters and add to export request if not null
+      String[] kinds = request.getParameterValues("kind");
+      if (kinds != null) {
+        JSONArray kindsJson = new JSONArray(kinds);
+        entityFilter.put("kinds", kinds);
+      }
+
+      // Read namespace parameters and add to export request if not null
+      String[] namespaces = request.getParameterValues("namespace_id");
+
+      if (namespaces != null) {
+        JSONArray namespacesJson = new JSONArray(namespaces);
+        entityFilter.put("namespaceIds", namespacesJson);
+      }
+
+      // Add entity filter to payload
+      // Finish export request payload
+      exportRequest.put("entityFilter", entityFilter);
+
+      // Send export request
+      OutputStreamWriter writer = new OutputStreamWriter(connection.getOutputStream());
+      exportRequest.write(writer);
+      writer.close();
+
+      // Examine server's response
+      if (connection.getResponseCode() != HttpURLConnection.HTTP_OK) {
+        // Request failed, report and log errors
+        InputStream s = connection.getErrorStream();
+        InputStreamReader r = new InputStreamReader(s, StandardCharsets.UTF_8);
+        String errorMessage =
+            String.format(
+                "got error (%d) response %s from %s",
+                connection.getResponseCode(), CharStreams.toString(r), connection.toString());
+        log.warning(errorMessage);
+        response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+        response.setContentType("text/plain");
+        response.getWriter().println(
+            "Failed to initiate export.");
+
+        
+      } else {
```

Comment:
This part could still use some work. You want the flow of logic to look like this:
```java
if (connection.getResponseCode() != HttpURLConnection.HTTP_OK)  {
    // handle unexpected error
    return
}
// Success, print export operation information
JSONObject exportResponse = new JSONObject(new JSONTokener(connection.getInputStream()));
```

This is easier to read because it more clearly indicates "If thing is bad, handle" and then the "thing is good" section is the default path, instead of just another branch of the if statement.